### PR TITLE
Add resource to execute groovy file against Jenkins master

### PR DESCRIPTION
**Pull request by castorls**
In our chef recipe, we need to execute a groovy file and the only available resource permit to execute groovy script from inside of the recipe.

I've so modified the recipe to permit to execute external fil against Jenkins master. Here's the pull request.

### CODE REVIEW

**Comment by chef-supermarket** [comment]
Hi. I am an automated pull request bot named Curry. There are commits in this pull request whose authors are not yet authorized to contribute to Chef Software, Inc. projects or are using a non-GitHub verified email address. To become authorized to contribute, you will need to sign the Contributor License Agreement (CLA) as an individual or on behalf of your company. [You can read more on Chef's blog.](https://www.chef.io/blog/2014/06/23/changes-to-the-contributor-license-agreement-process)

## Non-GitHub Verified Committers

There are 1 commit author(s) whose commits are authored by a non-GitHub verified email address. Chef will have to manually verify that they are authorized to contribute.

[Please sign the CLA here.](https://supermarket.chef.io/become-a-contributor)

File: libraries/_executor.rb [hunk 1]
```diff
@@ -143,6 +143,20 @@ def groovy(script)
...
+    # Same as {Executor#groovy!}, but with the script file as parameter.
+    #
+    # @see groovy!
+    #
+    def groovyFile!(script)
+      scriptFile = File.open(script)
+      execute!("groovy #{scriptFile.path}")
+    ensure
+      scriptFile.close if scriptFile
+    end
```
Comment (jtimberman) [inline libraries/_executor.rb]: In Ruby programs, snake_case is preferred to camelCase for methods and variables. These should be

* `groovy_file`
* `script_file`

Respectively.

File: libraries/matchers.rb [hunk 2]
```diff
@@ -111,6 +111,13 @@ def execute_jenkins_script(resource_name)
       resource_name)
   end
 
+  def execute_jenkins_scriptFile(resource_name)
+    ChefSpec::Matchers::ResourceMatcher.new(
+      :jenkins_scriptFile,
+      :execute,
+      resource_name)
+  end
```
Comment (jtimberman) [inline libraries/matchers.rb]: per above comment, this should be `execute_jenkins_script_file` and `:jenkins_script_file`.

File: libraries/scriptFile.rb [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+# limitations under the License.
+#
+
+require_relative 'command'
+require_relative '_params_validate'
+
+class Chef
+  class Resource::JenkinsScriptFile < Resource::JenkinsCommand
+    # Chef attributes
+    provides :jenkins_scriptFile
```
Comment (jtimberman) [inline libraries/scriptFile.rb]: `:jenkins_script_file`

File: libraries/scriptFile.rb [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+    # Set the resource name
+    self.resource_name = :jenkins_scriptFile
```
Comment (jtimberman) [inline libraries/scriptFile.rb]: `:jenkins_script_file`

File: libraries/scriptFile.rb [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+    action(:execute) do
+      converge_by("Execute script #{new_resource}") do
+        executor.groovyFile!(new_resource.file)
+      end
+    end
+  end
+end
+
+Chef::Platform.set(
+  resource: :jenkins_scriptFile,
```
Comment (jtimberman) [inline libraries/scriptFile.rb]: `:jenkins_script_file`

File: libraries/scriptFile.rb [hunk 6]
```diff
@@ -0,0 +1,62 @@
+#
+# Cookbook Name:: jenkins
+# HWRP:: scriptFile
```
Comment (jtimberman) [inline libraries/scriptFile.rb]: The filename should be `libraries/script_file.rb` and this comment `# Libraries: script_file_`

File: README.md [hunk 7]
```diff
@@ -93,6 +93,16 @@ jenkins_script 'add_authentication' do
 end
 ```
 
+### jenkins_scriptFile 
+This resource executes external groovy file against the Jenkins master. By the nature of this command, it is **not** idempotent.
+The file can be a resource of the recipe
+```ruby
+jenkins_scriptFile "initJenkins" do
+  file "/tmp/initJenkins.groovy"  
+end
```
Comment (jtimberman) [inline README.md]: `jenkins_script_file` here per other comments in this PR.

**Comment by jtimberman** [comment]
Thank you for this pull request. I think it's probably in line with the "resource primitives" focus of this cookbook, @castorls This needs to be rebased with master, and we don't have a CLA on file. See the message from Supermarket's curry bot above for more information.

**Comment by schisamo** [comment]
Closing this for now as the author has not responded for 6 months. Feel free to open a new PR.

**Pull request closed without merging**